lib/switchtec.c

```diff
@@ -147,6 +147,25 @@ static int set_gen_variant(struct switchtec_dev * dev)
 	return 0;
 }
 
+static int set_local_pax_id(struct switchtec_dev *dev)
+{
+	unsigned char local_pax_id;
+	int ret;
+
+	dev->local_pax_id = -1;
+
+	if (!switchtec_is_pax(dev))
+		return 0;
+
+	ret = switchtec_cmd(dev, MRPC_GET_PAX_ID, NULL, 0,
+			    &local_pax_id, sizeof(local_pax_id));
+	if (ret)
+		return -1;
+
+	dev->local_pax_id = local_pax_id;
```

Comment:
Honestly, I don't like this. I'd rather just stick with the constant SWITCHTEC_PAX_ID_LOCAL. If the user specifies the local pax_id on the command line then it will still use MRPC commands. It will only fall back to the default behavior if the pax id is not actually specified.

Reply:
I think there's still some benefit if we take '-x local_pax_id' the same as '-x' is not specified. 

- When do scripting, user maybe not aware of what the pax id of local pax is, but just specifying all the valid pax id in the script. I think it would be more reasonable to check for local/remote, rather then '-x' parameter.
- The new local_pax_id is also meaningful for some cases that user want to know on which pax switch they are locally operating on and therefore apply some policy (the API to get it is not available now). 

User would expect the behavior is consistent if the target is local, no matter if the '-x' is specified or not. Therefore the rule is also clear, when accessing local, use default ops and when accessing remote, use MRPC ones.

Reply:
I don't know. I see it as more consistent to behave one way if -x is specified and another way if it's not. It's weird to have some -x *values* do one thing and others and others do another.

Reply:
It seems to me that the more fundamental difference is local and remote. The users should have the sense that for a PAX fabric, they see local switch and remote switches, and therefore the ops for local and ops for remote. So I'd think current policy is more consistent with the concept that the firmware wants to deliver to users.

Reply:
Whatever, they *should* be consistent anyway.